### DISCUSSION TITLE
Base64 encoded SVG image is being identified as PNG

### DISCUSSION THREAD

**Discussion by azakero** [symfony/symfony#53351] [Q&A]
I am trying to add a validation in my Symfony controller where I need to validate file types of base64 image. So basically I am sending base64 images to the backend and there I am checking whether the file is jpeg or png. If the file is anything other than those then the validation should fail.

So my code seems to be working fine for JPEG and PNG where it passes the validation and fails for WebP. But when I try with a base64 SVG file, it passes as well (which should not). If I debug it, I can see the mime type of the file as image/png. So apparently it is being identified as a PNG file.

Here's the minimum reproducible example:

```
public function isSvg(array $file): bool
{
    $base64Image  = $file['base64'];

    $binaryData = base64_decode($base64Image);

    $directoryPath = '/tmp';

    if (!is_dir($directoryPath)) {
      mkdir($directoryPath, 0777, true);
    }

    $tempFileName = tempnam($directoryPath, 'uploaded_file');

    file_put_contents($tempFileName, $binaryData);

    $file = new File($tempFileName);

    $mimeType = $file->getMimeType();
}
```

Please note that, mimetype shows correctly if we use a svg file directly from the filesystem instead of base64 svg. 

What could be the issue here?

**Comment by MatTheCat** [comment]
Could not reproduce on a random SVG file using the `FileinfoMimeTypeGuesser`.

Please tell us which guesser is used in your case and provide a problematic SVG file.
